File: CustomerOrderService/app/Http/Responses/ApiResponse.php
<?php

namespace App\Http\Responses;

use Illuminate\Http\JsonResponse;

class ApiResponse
{
    public static function success(mixed $data, string $message = '', int $status = 200): JsonResponse
    {
        return response()->json([
            'success' => true,
            'data'    => $data,
            'message' => $message,
            'errors'  => [],
        ], $status);
    }

    public static function error(string $message, array $errors = [], int $status = 400): JsonResponse
    {
        return response()->json([
            'success' => false,
            'data'    => null,
            'message' => $message,
            'errors'  => $errors,
        ], $status);
    }
}

File: AuthService/src/AuthService.Shared/Wrappers/ApiResponse.cs
namespace AuthService.Shared.Wrappers;

/// <summary>Envelope estándar para todas las respuestas de la API.</summary>
public class ApiResponse<T>
{
    public bool Success { get; init; }
    public T? Data { get; init; }
    public string Message { get; init; } = string.Empty;
    public IEnumerable<string> Errors { get; init; } = [];

    /// <summary>Crea una respuesta de éxito con datos.</summary>
    public static ApiResponse<T> Ok(T data, string message) =>
        new() { Success = true, Data = data, Message = message, Errors = [] };

    /// <summary>Crea una respuesta de error con mensaje y lista de errores opcional.</summary>
    public static ApiResponse<T> Fail(string message, IEnumerable<string>? errors = null) =>
        new() { Success = false, Data = default, Message = message, Errors = errors ?? [] };
}
